### PR TITLE
Migrating evoli to latest Amethyst master

### PR DESCRIPTION
**Pull request by sunreef**
I upgraded Amethyst version to the latest master.

The game runs fine. Prefabs and code have been updated.

Some functionalities haven't been restored yet: the inspector and the amethyst-imgui are commented out for now.

### CODE REVIEW

File: src/components/creatures.rs [hunk 1]
```diff
@@ -76,8 +77,8 @@ impl Wander {
 #[serde(deny_unknown_fields)]
 pub struct CreaturePrefabData {
     pub name: Option<Named>,
-    graphics: Option<GraphicsPrefab<Vec<PosNormTex>, ObjFormat, TextureFormat>>,
-    movement: Option<Movement>,
+    graphics: Option<GraphicsPrefab<Vec<PosNormTex>>>,
```
Comment (Frizi) [inline src/components/creatures.rs]: `PosNormTex` is not used by renderer now, as we've migrated to separate buffers in all builtin 3d passes. You have to use at least `GraphicsPrefab<(Vec<Position>, Vec<TexCoord>)>` for `Flat` renderer to pay attention to those objects.

File: src/render_graph.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        );
+
+        let sprite = graph_builder.add_node(
+            SubpassBuilder::new()
+                .with_group(DrawFlatDesc::new().builder())
+                .with_color(color)
+                .with_depth_stencil(depth)
+                .into_pass(),
+        );
+//        let sprite_trans = graph_builder.add_node(
```
Comment (Frizi) [inline src/render_graph.rs]: even if you add multiple rendergroups, those should usually be in single pass unless there is good reason to split. In just means do multiple `with_group` calls one after another.

File: src/render_graph.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        let depth = graph_builder.create_image(
+            window_kind,
+            1,
+            Format::D32Sfloat,
+            Some(ClearValue::DepthStencil(ClearDepthStencil(1.0, 0))),
+        );
+
+        let sprite = graph_builder.add_node(
+            SubpassBuilder::new()
+                .with_group(DrawFlatDesc::new().builder())
```
Comment (Frizi) [inline src/render_graph.rs:82]: If you also want to render UI and sprites, add
```rust
.with_group(DrawFlat2DDesc::new().builder())
.with_group(DrawFlat2DTransparentDesc::new().builder())
.with_group(DrawUiDesc::new().builder())
```

File: src/render_graph.rs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        );
+
+        let depth = graph_builder.create_image(
+            window_kind,
+            1,
+            Format::D32Sfloat,
+            Some(ClearValue::DepthStencil(ClearDepthStencil(1.0, 0))),
+        );
+
+        let sprite = graph_builder.add_node(
```
Comment (Frizi) [inline src/render_graph.rs]: rename `sprite` to `main_pass`, there is no reason to even suggest splitting passes by some object type.

File: Cargo.toml [hunk 5]
```diff
@@ -5,13 +5,12 @@ authors = []
...
-amethyst = "0.10.0"
-amethyst_derive = "0.3.0"
-amethyst_test = "0.1.0"
+amethyst = { git = "https://github.com/amethyst/amethyst.git", features = ["vulkan"]}
+#amethyst_test = "0.1.0"
 rand = "0.6.5"
 log = "0.4.6"
-amethyst-imgui = { git = "https://github.com/awpteamoose/amethyst-imgui.git", branch = "amethyst-v0.10" }
-amethyst-inspector = { git = "https://github.com/awpteamoose/amethyst-inspector.git", branch = "amethyst-v0.10" }
+#amethyst-imgui = { git = "https://github.com/awpteamoose/amethyst-imgui.git"}
```
Comment (jaynus) [inline Cargo.toml]: I've released 0.2 of amethyst-imgui which supports the latest master of amethyst.  Inspector may not support it, but that should be an easy forward port.

File: src/components/creatures.rs (hunk 1 above)
Comment (sunreef) [reply src/components/creatures.rs]: Thanks for the info. This fixed the rendering of the sprites.
However, now they appear upside down. Does Rendy use a different convention for axes than the previous rendering system ?

**Pull request merged**